processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -409,14 +420,28 @@ public String apply(@Nullable String input)
           @Nullable ArrayList<Iterable<Rowboat>> boats
       )
       {
-        return CombiningIterable.create(
-            new MergeIterable<Rowboat>(
-                Ordering.<Rowboat>natural().nullsFirst(),
-                boats
-            ),
-            Ordering.<Rowboat>natural().nullsFirst(),
-            new RowboatMergeFunction(sortedMetricAggs)
-        );
+        if (rollup) {
+          return CombiningIterable.create(
+              new MergeIterable<Rowboat>(
+                  Ordering.<Rowboat>natural().nullsFirst(),
+                  boats
+              ),
+              Ordering.<Rowboat>natural().nullsFirst(),
+              new RowboatMergeFunction(sortedMetricAggs)
+          );
+        } else {
+          return new MergeIterable<Rowboat>(
+              new Ordering<Rowboat>()
+              {
+                @Override
+                public int compare(Rowboat left, Rowboat right)
+                {
+                  return Longs.compare(left.getTimestamp(), right.getTimestamp());
```

Comment:
added. @jon-wei 
To pass SelectQueryRunnerTest, I changed ConcurrentLinkedQueue to ConcurrentLinkedDeque to use queue.descendingIterator()